### PR TITLE
added flextesa script only for drain-delegate and ballot ops

### LINKED ISSUE #2851: Update flextesa ci script
With flextesa no longer having new docker image for new protocols, we'd like to target the latest relevant docker image (nairobi oxheadalpha/flextesa:20230607) to at least run drain_delegate and ballot tests in our ci script `.github/workflows/main.yml`. 
Also, update the integration-tests/README.md #taquito-integration-tests-with-flextesa accordingly

### PR DESCRIPTION
**Pull request by dsawali**
closes #2851 

Updated `main.yml` to have a flextesa box running on Nairobinet (even though outdated), just to run ballot and drain delegate tests

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have run the linter against the changes
- [ ] You have added unit tests (if relevant/appropriate)
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [ ] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [ ] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/config.ts [hunk 1]
```diff
@@ -140,6 +141,15 @@ const oxfordnetEphemeral: Config =
 const oxfordnetSecretKey: Config =
   { ...oxfordnetEphemeral, ...{ signerConfig: defaultSecretKey }, ...{ defaultRpc: 'http://ecad-oxfordnet-full:8732' } };
 
+const nairobinetSecretKey: Config =
```
Comment (hui-an-yang) [inline integration-tests/config.ts:144]: perhaps the naming of the constant can be something like nairobiSandbox / nairobiFlextesa? think it'll be helpful to distinguish from testnet

File: integration-tests/config.ts [hunk 2]
```diff
@@ -140,6 +141,15 @@ const oxfordnetEphemeral: Config =
 const oxfordnetSecretKey: Config =
   { ...oxfordnetEphemeral, ...{ signerConfig: defaultSecretKey }, ...{ defaultRpc: 'http://ecad-oxfordnet-full:8732' } };
 
+const nairobinetSecretKey: Config = 
+  defaultConfig({
+    networkName: 'NAIROBINET',
+    protocol: Protocols.PtNairobi,
+    defaultRpc: 'http://ecad-nairobinet-full:8732',
```
Comment (hui-an-yang) [inline integration-tests/config.ts:148]: defaultRpc: 'http://localhost:20000'

File: integration-tests/config.ts [hunk 3]
```diff
@@ -172,6 +182,8 @@ if (process.env['RUN_WITH_SECRET_KEY']) {
   providers.push(oxfordnetSecretKey);
 } else if (process.env['RUN_GHOSTNET_WITH_SECRET_KEY']) {
   providers.push(ghostnetSecretKey);
+} else if(process.env['RUN_NAIROBINET_WITH_SECRET_KEY']) {
```
Comment (hui-an-yang) [inline integration-tests/config.ts:185]: perhaps RUN_NAIROBIBOX_WITH_SECRET_KEY ?

File: integration-tests/package.json [hunk 4]
```diff
@@ -5,6 +5,7 @@
     "test:secret-key": "RUN_WITH_SECRET_KEY=true jest --runInBand",
     "test:oxfordnet": "OXFORDNET=true jest",
     "test:oxfordnet-secret-key": "RUN_OXFORDNET_WITH_SECRET_KEY=true jest --runInBand",
+    "test:nairobinet-secret-key": "RUN_NAIROBINET_WITH_SECRET_KEY=true jest --runInBand",
```
Comment (hui-an-yang) [inline integration-tests/package.json:8]: nairobibox-secret-key?

File: package.json [hunk 5]
```diff
@@ -20,7 +20,7 @@
     "example": "cd example && npm run example",
     "build-docs": "typedoc --tsconfig ./tsconfig.typedoc.json",
     "gh-pages": "gh-pages",
-    "integration-tests": "npm -w integration-tests run test:originate-known-contracts && npm -w integration-tests run test:secret-key -- --testPathIgnorePatterns ledger-signer-failing-tests.spec.ts ledger-signer.spec.ts contract-estimation-tests.spec.ts rpc-get-protocol-constants.spec.ts",
+    "integration-tests": "npm -w integration-tests run test:originate-known-contracts && npm -w integration-tests run test:nairobinet-secret-key -- --testPathIgnorePatterns ledger-signer-failing-tests.spec.ts ledger-signer.spec.ts contract-estimation-tests.spec.ts rpc-get-protocol-constants.spec.ts",
```
Comment (hui-an-yang) [inline package.json:23]: I suppose running integration tests from root folder, shouldn't target nairobinet only

File: integration-tests/sandbox-env.sh [hunk 6]
```diff
@@ -1,8 +1,7 @@
 #!/bin/sh
-export TEZOS_NETWORK_TYPE=sandbox
```
[the remark this answers is not in the document]
Comment (hui-an-yang) [reply integration-tests/sandbox-env.sh:2]: is there particular reason it's taken out?

**Review by hui-an-yang — CHANGES_REQUESTED** [review]
Namings are suggestions only; will respect your decision of what to keep 
Just some small comments and would like to remind updating integration-tests/README.md #taquito-integration-tests-with-flextesa section

File: integration-tests/config.ts (hunk 1 above)
Comment (dsawali) [reply integration-tests/config.ts:144]: the 'secretKey' part implies that its running on a sandbox I think Daniele or Roxane created the term, I don't have any problems with the current naming scheme.

File: integration-tests/config.ts (hunk 3 above)
Comment (dsawali) [reply integration-tests/config.ts:185]: nairobinet_secret_key implies a Nairobi network running using a secret key (which in this case implies the usage of a sandbox).

either we keep the name, or just 'RUN_NAIROBIBOX'. Which one do you like better?

File: integration-tests/sandbox-env.sh (hunk 6 above)
[the remark this answers is not in the document]
Comment (hui-an-yang) [reply integration-tests/sandbox-env.sh:2]: wait I think you have a point maybe removing all TEZOS_NETWORK_TYPE is cleaner

File: integration-tests/config.ts (hunk 2 above)
Comment (dsawali) [reply integration-tests/config.ts:148]: it actually doesn't matter since it just acts like a placeholder, but I'll update it

**Pull request merged**